VideoLocker/src/main/java/org/edx/mobile/view/RegisterActivity.java

```diff
@@ -135,79 +134,55 @@ public void onClick(View v) {
-                if (field.getFieldType().equals(RegistrationFieldType.CHECKBOX)
-                        && field.getAgreement() != null) {
-                    // this is agreement field
-                    // this must be added at the end of the form
-                    // hold on it
-                    agreements.add(field);
-                } else {
-                    IRegistrationFieldView fieldView = IRegistrationFieldView.Factory.getInstance(inflater, field);
-                    if (fieldView != null) mFieldViews.add(fieldView);
+                for (RegistrationFormField field : registrationDescription.getFields()) {
+                    if (IRegistrationFieldView.HONOR_CODE_CHECKBOX_ID.equals(field.getName())) {
+                        agreements.add(field);
+                    } else {
+                        IRegistrationFieldView fieldView = IRegistrationFieldView.Factory.getInstance(inflater, field);
+                        if (fieldView != null) mFieldViews.add(fieldView);
+                    }
                 }
-            }
 
-            // add required and optional fields to the window
-            for (IRegistrationFieldView v : mFieldViews) {
-                if (v.getField().isRequired()) {
-                    requiredFieldsLayout.addView(v.getView());
-                } else {
-                    optionalFieldsLayout.addView(v.getView());
+                // add required and optional fields to the window
+                for (IRegistrationFieldView v : mFieldViews) {
+                    if (v.getField().isRequired()) {
+                        requiredFieldsLayout.addView(v.getView());
+                    } else {
+                        optionalFieldsLayout.addView(v.getView());
+                    }
                 }
-            }
 
-            // add agreement fields to the window if available
-            for (RegistrationFormField agreement : agreements) {
-                IRegistrationFieldView agreementView = IRegistrationFieldView.Factory.getInstance(inflater, agreement);
-                agreementView.setActionListener(new IRegistrationFieldView.IActionListener() {
-                    @Override
-                    public void onClickAgreement(RegistrationAgreement agreement) {
-                        showAgreement(agreement);
-                    }
-                });
-                agreementLayout.addView(agreementView.getView());
+                // add agreement fields to the window if available
+                for (RegistrationFormField agreement : agreements) {
+                    IRegistrationFieldView agreementView = IRegistrationFieldView.Factory.getInstance(inflater, agreement);
+                    agreementView.setActionListener(new IRegistrationFieldView.IActionListener() {
+                        @Override
+                        public void onClickAgreement() {
+                            showAgreement();
```

Comment:
This function only has 1 line in it and is being called from here alone, so, that line should be moved here and the function should be removed.